src/contrasts.jl

```diff
@@ -79,25 +79,25 @@ termnames(C::MyCoding, levels, baseind) = ...
 ```
 
 """
-abstract AbstractContrasts
+abstract type AbstractContrasts end
 
 # Contrasts + Levels (usually from data) = ContrastsMatrix
-type ContrastsMatrix{C <: AbstractContrasts, T}
+mutable struct ContrastsMatrix{T}
```

Comment:
This shouldn't be changed as it makes the type non-concrete. Only remove type parameters when they are used only once in a function's signature (if they are used twice, they enforce consistency of the particular concrete type which is passed). Same below in several places.